addons/bus/static/src/js/services/web_bus_service.js

```diff
@@ -0,0 +1,93 @@
+/** @odoo-module **/
+
+import { serviceRegistry } from "@web/webclient/service_registry";
+import { legacySetupProm } from "@web/legacy/legacy_setup";
+
+export const webBusService = {
+    async deploy(env) {
+        const _assets = {};
+        let _assetsChangedNotificationId = null;
+        let _bundleNotifTimerID = null;
+        const legacyEnv = await legacySetupProm;
```

Comment:
The spirit is here, obviously, but we aim at deploying services as fast as possible
In main.js we only wait for stuff that we are obligated to (and we'll try in the future to wait even less)

So, I think we may rely on the bus's event "WEB_CLIENT_READY" instead of waiting for promises, And it will be easier in tests I think